pixeltable/utils/arrow.py

```diff
@@ -0,0 +1,112 @@
+    elif isinstance(arrow_type, pa.FixedShapeTensorType):
+        assert arrow_type.value_type in _pa_to_pt, f'unknown value type {arrow_type.value_type} for array'
+        return ts.ArrayType(shape=arrow_type.shape, dtype=get_pixeltable_type(arrow_type.value_type))
+    else:
+        assert False, f'default conversion for {arrow_type} not implemented'
+
+def get_arrow_type(pixeltable_type : ts.ColumnType) -> pa.DataType:
+    if pixeltable_type.__class__ in _pt_to_pa:
+        return _pt_to_pa[pixeltable_type.__class__]
+    elif isinstance(pixeltable_type, ts.ArrayType):
+        return pa.fixed_shape_tensor(pa.from_numpy_dtype(pixeltable_type.numpy_dtype()), pixeltable_type.shape)
+    else:
+        assert False, f'default conversion for {pixeltable_type} not implemented'
+
+def get_pixeltable_schema(arrow_schema : pa.Schema) -> Dict[str, ts.ColumnType]:
+    return {field.name: get_pixeltable_type(field.type) for field in arrow_schema}
+
+def get_arrow_schema(pixeltable_schema : Dict[str, Any]) -> pa.Schema:
+    return pa.schema({name: get_arrow_type(dtype) for name, dtype in pixeltable_schema.items()})
+
+def to_pydict(batch : pa.RecordBatch) -> Dict[str, Iterable[Any]]:
+    """ Convert a RecordBatch to a dictionary of lists, unlike pa.lib.RecordBatch.to_pydict,
+        this function will not convert numpy arrays to lists, and will preserve the original numpy dtype.
+    """
+    out = {}
+    for k, name in enumerate(batch.schema.names):
+        col = batch.column(k)
+        if isinstance(col.type, pa.FixedShapeTensorType):
+            # treat array columns as numpy arrays to easily preserve numpy type
+            out[name] = col.to_numpy(zero_copy_only=False)
+        else:
+            # for the rest, use pydict to preserve python types
+            out[name] = col.to_pylist()
+
+    return out
+
+def iter_tuples(batch : pa.RecordBatch) -> Iterator[Dict[str,Any]]:
+    """ Convert a RecordBatch to an iterator of dictionaries. also works with pa.Table and pa.RowGroup """
+    pydict = to_pydict(batch)
+    n = len(pydict[list(pydict.keys())[0]])
+    for i in range(n):
+        yield {col_name: values[i] for col_name, values in pydict.items()}
+
+
+def create_table_from_parquet(cl : 'pixeltable.Client',
+                              path_str: str,
+                              parquet_path: str,
+                              explain_only : bool = False,
+                              primary_key: Union[str, List[str]] = [],
+                              num_retained_versions: int = 10) -> Optional['catalog.InsertableTable']:
+    """ See pixeltable.Client.create_table_from_parquet for documentation
+    """
+
+    input_path = pathlib.Path(parquet_path).expanduser()
+    parquet_dataset = pa.parquet.ParquetDataset(input_path)
+    arrow_schema = parquet_dataset.schema
+    base_schema = get_pixeltable_schema(arrow_schema)
+
+    print('mapping parquet columns to pixeltable schema:', base_schema)
```

Comment:
Remove `print` statement (or replace with logs?)

Reply:
logs get sent many places, but this is meant to be user visible, letting them decide if the mapping works for them.

Reply:
This speaks to a broader question about the intended usage patterns for pixeltable. Are we expecting it to be used interactively? Or programmatically as part of a service that does (say) parquet importing on a schedule or as an event trigger? Or both?

I generally cringe when I see print statements in production code, but we can discuss (and it'd be good to get @mkornacker 's thoughts too).